### PR TITLE
[DCOS-50422] fix: clean orphaned versions of UI on startup, reset, and updating

### PR DESCRIPTION
**Pull request by brandonc**
Delete orphaned versions of the UI when starting up, resetting, and updating

## Testing

1. set CLUSTER_URL and AUTH_TOKEN
2. docker-compose up
3. run `docker exec -it <update serivice container id> bash` and change to versions-root directory: `cd /opt/mesosphere/active/versions/`
4. update to a new version using `curl -X POST http://localhost:5000/api/v1/update/2.blah.0/`, observe that the old version is removed, the symlink at `/opt/mesosphere/active/dsos-ui-dist` is updated to the new version
5. reset using `curl -X DELETE http://localhost:5000/api/v1/reset/`, observe that all versions are removed and the symlink is pointing to `/opt/mesosphere/active/dcos-ui`

### CODE REVIEW

File: uiService/api.go [hunk 1]
```diff
@@ -133,47 +133,46 @@ func resetToDefaultUIHandler(service *UIService) func(http.ResponseWriter, *http
...
-		logrus.WithField("CurrentVersion", currentVersion).Debug("Received reset request.")
 
-		if updatingVersion, lockErr := setServiceUpdating(service, ""); lockErr != nil {
-			var message string
-			if UIVersion(updatingVersion) == PreBundledUIVersion {
-				message = "Cannot process reset, another reset is currently in progress."
-			} else {
-				message = "Cannot process reset, an update is currently in progress."
+		if UIVersion(currentVersion) != PreBundledUIVersion {
+			logrus.WithField("CurrentVersion", currentVersion).Debug("Received reset request.")
```
Comment (TattdCodeMonkey) [inline uiService/api.go]: We should move this through line 154 `defer resetServiceFromUpdate(service)` to above and outside this `if`. We would want the service set to updating while we clear the versions root.

**Comment by DanielMSchmidt** [comment]
I got this error:

```
dcos-ui_1    | time="2019-03-25T12:56:01Z" level=error msg="Could not remove all versions." error="lstat /opt/mesosphere/active/versions: no such file or directory"
dcos-ui_1    | time="2019-03-25T12:56:01Z" level=fatal msg="Failed to initiate ui service" error="failed to clean up unused versions: Failed to remove all versions"
```

File: uiService/root.go [hunk 2]
```diff
@@ -53,8 +53,11 @@ func SetupService(cfg *config.Config) (*UIService, error) {
 	}
 
 	checkUIDistSymlink(cfg)
-	checkVersionsRoot(cfg)
```
Comment (TattdCodeMonkey) [inline uiService/root.go]: why did you remove the `checkVersionsRoot` call?

File: updateManager/client.go [hunk 3]
```diff
@@ -216,6 +219,43 @@ func (um *Client) UpdateToVersion(version string, updateCompleteCallback func(st
...
+func (um *Client) RemoveAllVersionsExcept(omitVersion string) error {
+	var err error
+
+	root := um.Config.VersionsRoot()
+	if len(root) <= 1 {
+		logrus.WithField("versions-root", root).Fatal("Potentially dangerous versions-root configuration.")
+		return ErrRemovingAllVersions
+	}
+
+	err = afero.Walk(um.Fs, root, func(path string, info os.FileInfo, walkErr error) error {
```
Comment (TattdCodeMonkey) [inline updateManager/client.go]: `Walk` recursively walks down the version root directory, I think a `ReadDir` maybe better here.

```golang
dirContent, err := afero.ReadDir(um.Fs, root)
if err != nil {
  // handle error
}
for _, info := range dirContent {
  if info.IsDir() && info.Name() != omitVersion {
    um.RemoveVersion(info.Name())
  }
}
```

Comment (brandonc) [reply updateManager/client.go]: I misread the docs 😳

File: uiService/root.go (hunk 2 above)
Comment (brandonc) [reply uiService/root.go]: I think I was planning on doing versionsRoot startup in a single method but changed my mind and forgot to put it back!

**Comment by brandonc** [comment]
I think the error occured if the versions root does not exist because I removed `checkVersionsRoot` 🤦‍♂️

**Comment by brandonc** [comment]
@TattdCodeMonkey fixed and rebased

**Comment by DanielMSchmidt** [comment]
@brandonc can this be merged now?

**Comment by TattdCodeMonkey** [comment]
@DanielMSchmidt yes :)
